### PR TITLE
WIP - Voter history for parties and addresses

### PR DESCRIPTION
**Pull request by slavingia**

### CODE REVIEW

**Comment by j-ro** [comment]
Looks pretty good to me! Though not sure about parties as an array vs an object, though I guess folks could have more than one party as seen via time?

**Comment by slavingia** [comment]
Yep, I think a common use case for this will be to monitor party switches over time. I also think it'll be useful to have objects so that we have consistent party naming, and potentially a mapping of parties in each state with the major national parties, as they sometimes differ.

**Comment by j-ro** [comment]
Looks good to me! Maybe time to put these new fields on the examples?

File: people.md [hunk 1]
```diff
@@ -59,7 +59,8 @@ A list of fields specific to the Person resource.
 |honorific_suffix           |string    |An honorific suffix like "Jr.", "Ph.D", etc...
 |gender         |enum     |The gender binary with which a person most closely identifies, or "Other" if the person identifies with neither. One of "Female", "Male", or "Other".
 |gender_identity|string     |The self-described gender with which a person identifies. While this field is free-form, data should still follow standardized forms whenever possible (i.e. use "Female" and not "female" or "F"). _Examples: If a person self-identifies as "Female", both_ `gender` _and_ `gender_identity` _fields should have a value of "Female". If a person self-identifies as "Transgender Female",_ `gender` _should have a value of "Female" and_ `gender_identity` _should have a value of "Transgender Female"._
-|party_identification          |flexenum     |Flexenum describing the person's politcal party_identification. One of "None", "Democratic", "Republican", "Independent", or another free-form string.
+|party_identification   |flexenum     |Flexenum describing the person's politcal party_identification. One of "None", "Democratic", "Republican", "Independent", or another free-form string.
```
Comment (j-ro) [inline people.md]: probably should be party identification, not party_identification in the description?

Comment (j-ro) [inline people.md]: Also probably make clear that this is the current identification, as opposed to the parties hash array?

File: people.md [hunk 2]
```diff
@@ -59,7 +59,8 @@ A list of fields specific to the Person resource.
...
+|parties |[Parties[]](#parties)|An array of party object hashes associated with the person.
```
Comment (j-ro) [inline people.md]: make clear this is for historical purposes, to differentiate it from the above?

File: people.md [hunk 3]
```diff
@@ -88,6 +89,14 @@ These JSON hashes included in the table above are broken out into their own tabl
 |birthdate.day  |integer    |An integer representing the day of the birth date of the person.
 |birthdate.year|integer     |An integer representing the 4 digit year of the birth date of the person.
 
+#### Parties
+
+|Name          |Type      |Description
+|-----------    |-----------|--------------
+|parties.identification |flexenum |One of "None", "Democratic", "Republican", "Independent", or another free-form string.
+|parties.last_verified_date   |integer    |A value representing the last verified date of the party registration.
```
Comment (j-ro) [inline people.md]: type should be datetime

File: people.md (hunk 1 above)
Comment (slavingia) [inline people.md]: This would make it not backwards compatible

File: people.md (hunk 2 above)
Comment (slavingia) [inline people.md]: It would be historical and current in the hash though

**Comment by slavingia** [comment]
updated

File: people.md (hunk 1 above)
Comment (j-ro) [inline people.md]: I mean in the description part:

"Flexenum describing the person's politcal party_identification. One of "None", "Democratic", "Republican", "Independent", or another free-form string."

becomes

"Flexenum describing the person's politcal party identification. One of "None", "Democratic", "Republican", "Independent", or another free-form string."

File: people.md (hunk 2 above)
Comment (j-ro) [inline people.md]: then maybe have it say both current and historical -- just more description there on why this hash exists.

**Comment by slavingia** [comment]
cool, added

**Comment by j-ro** [comment]
Looks good to me!

**Comment by joshco** [comment]
Approved by CMTE 3/9/17

**Pull request merged**